### PR TITLE
Add String#rchop? and String#lchop?

### PR DESCRIPTION
**Pull request by j8r**
`String#rchop?` and `String#lchop?` are the nillable variants of `String#rchop` and `String#lchop`.

A use case for them is to remove a prefix/suffix when a string starts/ends with a certain string, else continue.

```cr
# This is inefficient because lchop also do a starts_with? check internally 
if mystring.starts_with? "prefix"
  substring = mystring.lchop "prefix"
# This is also not optimal and verbose
elsif (substring = mystring.lchop "prefix") != mystring
  substring
# The new way to do - try to lchop
elsif substring = mystring.lchop? "prefix"
  substring
end
```

### CODE REVIEW

**Comment by Sija** [comment]
Why not just calling `lchop/rchop`? You'd want to remove prefix/suffix anyway.

```cr
# This is inefficient because lchop also do a starts_with? check internally 
if mystring.starts_with? "prefix"
  substring = mystring.lchop "prefix"
# This is also not optimal and verbose
elsif (substring = mystring.lchop "prefix") != mystring
  substring
# The new way to do - try to lchop
elsif substring = mystring.lchop? "prefix"
  substring
end

# why all this dance if you can do:
mystring = mystring.lchop "prefix"
```

**Comment by j8r** [comment]
@Sija this are just examples, take this one:

```cr
FILE = <<-E
name=john
num=123
E

name = nil
num = nil
FILE.each_line do |line|
  if line.starts_with? "name="
    name = line.lchop "name="
  elsif line.starts_with? "num="
    num = line.lchop "num="
  end
end
p name, num
```

**Comment by Sija** [comment]
With such implementation I find the API confusing:

```cr
"foo".rchop("bar")  # => "foo"
"foo".rchop?("bar") # => nil
```

It's behavior is nothing like we know from from other `?` suffixed methods from stdlib - either query or nilable - and this one is neither of those.

**Comment by j8r** [comment]
The last commit brings improvements in the language to take advantage of this changes.

For me this methods are similar to `String#[index : Int]?` - try to perform the operation and return a value, else return nil.
Maybe that's the `rchop` and `lchop` behavior that is confusing, not `rchop?` and `lchop?`.

**Comment by j8r** [comment]
You're right @bcardiff , I naively modified the original methods and add new ones.

File: src/spec/source.cr [hunk 1]
```diff
@@ -15,10 +15,10 @@ module Spec
   # :nodoc:
   def self.relative_file(file)
     cwd = Dir.current
-    if file.starts_with?(cwd)
-      file = file[cwd.size..-1]
-      file = file[1..-1] if file.starts_with?('/')
+    if file_prefix = file.lchop? cwd
+      file = file_prefix.lchop '/'
```
Comment (asterite) [inline src/spec/source.cr]:
```sugestion
      file_prefix.lchop '/'
```

Comment (asterite) [inline src/spec/source.cr]:
```suggestion
      file_prefix.lchop '/'
```

Comment (j8r) [reply src/spec/source.cr]: I also renamed prefix to basename

**Pull request merged**